mount/mount_windows.go

```diff
@@ -97,32 +101,31 @@ func (mounter *Mounter) MountSensitive(source string, target string, fstype stri
 		getSMBMountMutex.LockKey(source)
 		defer getSMBMountMutex.UnlockKey(source)
 
-		var output string
-		var err error
 		username := allOptions[0]
 		password := allOptions[1]
-		if output, err = newSMBMapping(username, password, source); err != nil {
+		if output, err := newSMBMapping(username, password, source); err != nil {
 			klog.Warningf("SMB Mapping(%s) returned with error(%v), output(%s)", source, err, string(output))
 			if isSMBMappingExist(source) {
-				valid, errPath := isValidPath(source)
-				if errPath != nil {
-					return errPath
+				valid, err := isValidPath(source)
+				shouldRemovePath := (err == nil && !valid)
+				if err != nil && strings.Contains(strings.ToLower(err.Error()), accessDenied) {
```

Comment:
sorry I kind of misunderstand the logic a little bit before. So if no error, and path is not valid, it will remove mapping..

valid, err := isValidPath(source)
if !valid {
   if err == nil || isAccessDenied(err) {
       // perform remove mapping and create a new one
   } else {
      // log and return error
  }
}

func isAccessDenied (err error) bool {
    return strings.Contains(strings.ToLower(err.Error()), accessDenied)
}

Reply:
yes, if `!valid`, should still remove mapping

Reply:
there are two conditions that should remove mapping: 
 - `err == nil && !valid`
 -  `isAccessDenied(err)`

Reply:
isAccessDenied(err) also means !valid, right?

Reply:
```
valid, err := isValidPath(source)
if !valid {
   if err == nil || isAccessDenied(err) {
       // perform remove mapping and create a new one
   } else { // err might be temporal, return err and will retry later
      // log and return error
  }
} // if valid, do nothing
```

Reply:
thanks, I have made the changes, PTAL.